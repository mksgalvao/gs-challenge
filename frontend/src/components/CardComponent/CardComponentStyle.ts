import { createStyles, Theme, makeStyles } from "@material-ui/core/styles";

export const cardComponentStyles = makeStyles((theme: Theme) =>
  createStyles({
    cardRoot: {
      margin: 10,
      [theme.breakpoints.between("xs", "md")]: {
        alignItems: "center",
        width: 385,
      },
      [theme.breakpoints.up("md")]: {
        width: 360,
        alignItems: "center",
      },
      cursor: "pointer",
    },
    media: {
      height: 0,
      paddingTop: "56.25%", // 16:9
      hover: {
        "&:hover": {
          backgroundColor: theme.palette.success.main,
        },
      },
    },
    moreInfo: { marginLeft: "auto" },
    description: {
      marginTop: 10,
      height: 170,
    },
  })
);
