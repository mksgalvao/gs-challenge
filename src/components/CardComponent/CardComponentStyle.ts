import { createStyles, Theme, makeStyles } from "@material-ui/core/styles";

export const cardComponentStyles = makeStyles((theme: Theme) =>
  createStyles({
    cardsWrapper: {
      display: "flex",
      flexWrap: "wrap",

      [theme.breakpoints.between("xs", "sm")]: {
        flexDirection: "column !important",
        maxWidth: "100%",
      },
    },
    cardRoot: {
      maxWidth: 385,
      margin: 10,
      [theme.breakpoints.between("xs", "sm")]: {
        alignItems: "center",
      },
      cursor: "pointer",
    },
    media: {
      height: 0,
      paddingTop: "56.25%", // 16:9
      hover: {
        "&:hover": {
          backgroundColor: theme.palette.success.main,
        },
      },
    },
    moreInfo: { marginLeft: "auto" },
  })
);
