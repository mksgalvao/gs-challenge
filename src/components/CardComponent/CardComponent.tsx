import React from "react";

import Card from "@material-ui/core/Card";
import CardHeader from "@material-ui/core/CardHeader";
import CardMedia from "@material-ui/core/CardMedia";
import CardContent from "@material-ui/core/CardContent";
import CardActions from "@material-ui/core/CardActions";
import IconButton from "@material-ui/core/IconButton";
import { useHistory } from "react-router-dom";
import Typography from "@material-ui/core/Typography";
import EditIcon from "@material-ui/icons/Edit";
import DeleteIcon from "@material-ui/icons/Delete";

import { cardComponentStyles } from "./CardComponentStyle";

export const CardComponent = (props: any) => {
  const classes = cardComponentStyles();
  let history = useHistory();

  const onCardClick = (phone: any) => {
    history.push({
      pathname: `/info/${phone.name}`,
      state: { phone: phone },
    });
  };

  return (
    <div className={classes.cardsWrapper}>
      {props.products.map((product: any, index: number) => (
        <Card
          onClick={() => onCardClick(product)}
          key={index}
          className={classes.cardRoot}
        >
          <CardHeader title={product.name} subheader={product.manufacturer} />
          <CardMedia
            className={classes.media}
            image={product.imageFileName}
            title={product.price}
          />
          <CardContent>
            <Typography variant="body2" color="textSecondary" component="p">
              {product.description}
            </Typography>
          </CardContent>
          <CardActions disableSpacing>
            <IconButton aria-label="delete" onClick={props.onDelete}>
              <DeleteIcon />
            </IconButton>
            <IconButton aria-label="edit" onClick={props.onEdit}>
              <EditIcon />
            </IconButton>
          </CardActions>
        </Card>
      ))}
    </div>
  );
};
