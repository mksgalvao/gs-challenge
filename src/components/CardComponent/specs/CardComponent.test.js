import React from "react";
import enzyme from "enzyme";
import Adapter from "enzyme-adapter-react-16";
enzyme.configure({ adapter: new Adapter() });

import { shallow } from "enzyme";
import { createMemoryHistory } from "history";
import { CardComponent } from "../CardComponent";
import Card from "@material-ui/core/Card";

import IconButton from "@material-ui/core/IconButton";
import { PhonesList } from "../../../fixtures/phoneFixtures";

describe("Testing cardComponent", () => {
  const onDeletemock = jest.fn();
  const onEditmock = jest.fn();
  const onMoreInfosmock = jest.fn();

  let cardComponent;

  beforeEach(() => {
    cardComponent = shallow(
      <CardComponent
        products={PhonesList}
        onDelete={onDeletemock}
        onEdit={onEditmock}
      />
    );
  });

  it("CardComponent should render", () => {
    expect(cardComponent).toMatchSnapshot();
  }),
    it("Should call onDelete function when click in delete icon", () => {
      const deleteButton = cardComponent.find(IconButton).at(0);
      deleteButton.simulate("click");
      expect(onDeletemock).toHaveBeenCalledTimes(1);
    });
  it("Should call onEdit function when click in edit icon", () => {
    const editButton = cardComponent.find(IconButton).at(1);
    editButton.simulate("click");
    expect(onEditmock).toHaveBeenCalledTimes(1);
  });

  // TODO fix this test
  // it("Should call moreInfo function when click in more info button", () => {
  //   const history = createMemoryHistory();
  //   history.push("/");
  //   const moreInfos = cardComponent.find(Card).at(0);
  //   moreInfos.simulate("click");
  //   expect(history.push).toHaveBeenCalledWith("info/iPhone%207");
  // });
});
