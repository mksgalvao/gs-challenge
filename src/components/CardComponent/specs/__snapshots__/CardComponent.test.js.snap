// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing cardComponent CardComponent should render 1`] = `ShallowWrapper {}`;
